library/ssl_misc.h

```diff
@@ -1271,7 +1271,8 @@ static inline int mbedtls_ssl_write_handshake_msg( mbedtls_ssl_context *ssl )
 int mbedtls_ssl_finish_handshake_msg( mbedtls_ssl_context *ssl,
                                       size_t buf_len, size_t msg_len );
 
-int mbedtls_ssl_write_record( mbedtls_ssl_context *ssl, int force_flush );
+int mbedtls_ssl_write_record( mbedtls_ssl_context *ssl, int force_flush,
```

Comment:
The insertion of dummy CCSs in TLS 1.3 has been designed to avoid having to do something like that: they are sent before the first encrypted record. Thus here to avoid that we need to move the call to `mbedtls_ssl_set_outbound_transform` that is currently in `ssl_tls13_finalize_write_server_hello` to the beginning of `ssl_tls13_write_encrypted_extensions` and I think this should make the job.